src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksNewVendorPayeeAchServiceImpl.java

```diff
@@ -0,0 +1,485 @@
+            }
+            return false;
+        }
+    }
+    
+    private PaymentWorksVendor setStatusValuesForPaymentWorksNewVendorAchRequestedInKfs(PaymentWorksVendor pmwVendor) {
+        pmwVendor.setKfsAchProcessingStatus(PaymentWorksConstants.KFSAchProcessingStatus.ACH_REQUESTED);
+        return pmwVendor;
+    }
+    
+    private boolean pmwBankRoutingNumberIsValid(PaymentWorksVendor pmwVendor, List<String> errorMessages) {
+        if (StringUtils.isNumeric(pmwVendor.getBankAcctRoutingNumber())
+            && !(ObjectUtils.isNull(achBankService.getByPrimaryId(pmwVendor.getBankAcctRoutingNumber())))) {
+            return true;
+        } else {
+            errorMessages.add(getConfigurationService().getPropertyValueAsString(PaymentWorksKeyConstants.ERROR_BANK_ROUTING_NUMBER_INVALID));
+            return false;
+        }
+    }
+    
+    private boolean pmwBankAccountNumberIsValid(PaymentWorksVendor pmwVendor, List<String> errorMessages) {
+        if (StringUtils.isNumeric(pmwVendor.getBankAcctBankAccountNumber())) {
+            return true;
+        } else {
+            errorMessages.add(getConfigurationService().getPropertyValueAsString(PaymentWorksKeyConstants.ERROR_BANK_ACCOUNT_NUMBER_INVALID));
+            return false;
+        }
+    }
+    
+    private boolean pmwBankAccountTypeIsValid(PaymentWorksVendor pmwVendor, List<String> errorMessages) {
+        if (pmwBankAccountTypeIsDefinedInKfs(pmwVendor.getBankAcctType())) {
+            return true;
+        } else {
+            errorMessages.add(getConfigurationService().getPropertyValueAsString(PaymentWorksKeyConstants.ERROR_BANK_ACCOUNT_TYPE_INVALID));
+            return false;
+        }
+    }
+    
+    private boolean pmwBankAccountTypeIsDefinedInKfs(String pmwBankAccountTypeToVerify) {
+        List<PaymentWorksBankAccountType> matchingValues = getPaymentWorksBatchUtilityService().findAllPmwBankAccountTypesMatching(pmwBankAccountTypeToVerify);
+        return (matchingValues.size() == 1);
+    }
+    
+    private void performProcessingWhenInvalidAchAccountDataIsProvidedForApprovedKfsVendor(PaymentWorksVendor pmwVendor) {
+        updatePmwStagingTableWithKfsAchProcessingStatusForPmwApprovedKfsApprovedVendor(pmwVendor, PaymentWorksConstants.KFSAchProcessingStatus.ACH_REJECTED);
+        //Intentionally not making web service call to PMW as it is valid to have a KFS approved vendor with bad ACH data since ACH data is optional.
+    }
+    
+    private void performProcessingWhenAchAccountDataNotProvidedForApprovedKfsVendor(PaymentWorksVendor pmwVendor) {
+        updatePmwStagingTableWithKfsAchProcessingStatusForPmwApprovedKfsApprovedVendor(pmwVendor, PaymentWorksConstants.KFSAchProcessingStatus.NO_ACH_DATA);
+        //Intentionally not making web service call to PMW as it is valid for a vendor to not provide optional bank data.
+    }
+
+    private void performProcessingWhenNoKfsVendorIdentifersFoundForKfsApprovedVendor(PaymentWorksVendor pmwVendor) {
```

Comment:
Minor typo in method name. If we need to push this PR forward today, then I'm fine with deferring this fix to a future user story.